cmd/starter/c/starter.c

```diff
@@ -1169,7 +1145,68 @@ void load_overlay_module(void) {
             } else {
                 debugf("Overlay seems not supported by the kernel\n");
             }
-        };
+        }
+    }
+}
+
+/* read engine configuration from environment variables */
+static void read_engine_config(struct engine *engine) {
+    char *engine_config[MAX_ENGINE_CONFIG_CHUNK];
+    char env_key[ENGINE_CONFIG_ENV_PADDING] = {0};
+    char *engine_chunk;
+    unsigned long int nchunk = 0;
+    off_t offset = 0;
+    size_t length;
+    int i;
+
+    debugf("Read engine configuration\n");
+
+    engine_chunk = getenv(ENGINE_CONFIG_CHUNK_ENV);
+    if ( engine_chunk == NULL ) {
+        fatalf("No engine config chunk provided\n");
+    }
+    nchunk = strtoul(engine_chunk, NULL, 10);
+    if ( nchunk == 0 || nchunk > MAX_ENGINE_CONFIG_CHUNK ) {
+        fatalf("Bad number of engine config chunk provided '%s': 0 or > %d\n", engine_chunk, MAX_ENGINE_CONFIG_CHUNK);
+    }
+
+    for ( i = 0; i < nchunk; i++ ) {
+        /* safety check for env_key in case the hardcoded value is modified */
+        if ( snprintf(env_key, sizeof(env_key), ENGINE_CONFIG_ENV"%d", i+1) != ENGINE_CONFIG_ENV_PADDING-2 ) {
+            fatalf("Maximum engine config chunk >= 10\n");
```

Comment:
This error message / flow is a little confusing to me. If we have a hard limit on 9 as the `MAX_ENGINE_CONFIG_CHUNK`, and our `ENV_PADDING` needs to match  the env couldn't we instead...

- Compute the padding, rather than rely on it being hard coded
- Check earlier explicitly that `MAX_ENGINE_CONFIG_CHUNK <=9` 
- Check that `nchunk <= MAX_ENGINE_CONFIG_CHUNK`

I _think_ this would have the same effect, and it might make it easier to read the intent of the code?